### PR TITLE
Fix TidbMonitor Service Label

### LINKED ISSUE #2049: Prometheus service has no endpoint after deploying TidbMonitor
## Bug Report

**What version of Kubernetes are you using?**

1.12.8
**What version of TiDB Operator are you using?**

1.1.0-beta.2
**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

local-storage
**What's the status of the TiDB cluster pods?**

Running
**What did you do?**

https://github.com/pingcap/tidb-operator/issues/1891#issuecomment-604935411
**What did you expect to see?**
Prom service can be accessed via NodePort.
**What did you see instead?**
Prom service can not be accessed via NodePort.

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

Fix #2049
### What is changed and how does it work?
Change the label to of the TidbMonitor resources in order to make compatible with the helm 1.0 releases monitor. 

It would break the existed `TidbMonitor` (users have to delete the existed deployment manually or the prometheus service can't work), as `TidbMonitor` is not released yet, I think it's ok.

### Check List 

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
None
```

### CODE REVIEW

**Comment by cofyc** [comment]
`TiDBCluster TidbMonitor: Deploying and checking monitor` can't pass.

```
Stacktrace
/home/jenkins/agent/workspace/operator_ghpr_e2e_test_kind/go/src/github.com/pingcap/tidb-operator/tests/e2e/tidbcluster/tidbcluster.go:734
Mar 29 16:02:45.118: Expected
<string>:
to equal
<string>: tidb-operator
/go/pkg/mod/k8s.io/kubernetes@v1.16.0/test/e2e/framework/util.go:1339
Standard Output
[BeforeEach] [tidb-operator] TiDBCluster
/go/pkg/mod/k8s.io/kubernetes@v1.16.0/test/e2e/framework/framework.go:151
�[1mSTEP�[0m: Creating a kubernetes client
Mar 29 16:00:13.783: INFO: >>> kubeConfig: /etc/kubernetes/admin.conf
�[1mSTEP�[0m: Building a namespace api object, basename tidb-cluster
�[1mSTEP�[0m: Waiting for a default service account to be provisioned in namespace
[BeforeEach] [tidb-operator] TiDBCluster
/home/jenkins/agent/workspace/operator_ghpr_e2e_test_kind/go/src/github.com/pingcap/tidb-operator/tests/e2e/tidbcluster/tidbcluster.go:88
Mar 29 16:00:13.804: INFO: >>> kubeConfig: /etc/kubernetes/admin.conf
Mar 29 16:00:13.813: INFO: >>> kubeConfig: /etc/kubernetes/admin.conf
[It] TidbMonitor: Deploying and checking monitor
/home/jenkins/agent/workspace/operator_ghpr_e2e_test_kind/go/src/github.com/pingcap/tidb-operator/tests/e2e/tidbcluster/tidbcluster.go:734
Mar 29 16:01:14.398: INFO: >>> kubeConfig: /etc/kubernetes/admin.conf
Mar 29 16:01:14.463: INFO: >>> kubeConfig: /etc/kubernetes/admin.conf
Mar 29 16:02:14.408: INFO: >>> kubeConfig: /etc/kubernetes/admin.conf
Mar 29 16:02:14.481: INFO: >>> kubeConfig: /etc/kubernetes/admin.conf
Mar 29 16:02:14.801: INFO: >>> kubeConfig: /etc/kubernetes/admin.conf
Mar 29 16:02:45.118: INFO: Unexpected unequal occurred: and tidb-operator
Mar 29 16:02:45.118: FAIL: Expected
<string>:
to equal
<string>: tidb-operator
```

File: pkg/controller/pv_control.go [hunk 1]
```diff
@@ -135,7 +135,9 @@ func (rpc *realPVControl) UpdateMetaInfo(obj runtime.Object, pv *corev1.Persiste
 	pv.Labels[label.NamespaceLabelKey] = ns
 	pv.Labels[label.ComponentLabelKey] = component
 	pv.Labels[label.NameLabelKey] = pvc.Labels[label.NameLabelKey]
-	pv.Labels[label.ManagedByLabelKey] = pvc.Labels[label.ManagedByLabelKey]
+	// TidbMonitor's pod label don't have managedBy label in order to be compatible with 1.0 release monitor
+	// so we add manager-key label with `tidb-operator` values manually.
+	pv.Labels[label.ManagedByLabelKey] = "tidb-operator"
```
Comment (DanielZhangQD) [inline pkg/controller/pv_control.go]: Can we update the PVC to have this label? Otherwise, L125 needs to update.

Comment (Yisaer) [reply pkg/controller/pv_control.go]: updated.

Comment (DanielZhangQD) [reply pkg/controller/pv_control.go]: Then we can revert this change.

**Pull request merged**